File: packages/nitro-swift/Sources/NitroAvatar/AvatarGenderType.swift
/// Mirrors `AvatarGenderType` (packages/nitro-api/src/avatar/enum/AvatarGenderType.ts).
public enum AvatarGenderType: String, Sendable {
    case male = "M"
    case female = "F"
    case unisex = "U"

    public var numericValue: Int { self == .female ? 1 : 0 }

    public static func from(numericValue: Int) -> AvatarGenderType { numericValue == 1 ? .female : .male }
}

File: packages/nitro-api/src/avatar/enum/AvatarGenderType.ts
export enum AvatarGenderType {
    Male = 'M',
    Female = 'F',
    Unisex = 'U',
}

export class AvatarGenderTypeUtilities {
    public static genderToNumber(gender: AvatarGenderType) {
        switch (gender) {
            case AvatarGenderType.Female:
                return 1;
            default:
                return 0;
        }
    }

    public static numberToGender(gender: number) {
        switch (gender) {
            case 1:
                return AvatarGenderType.Female;
            default:
                return AvatarGenderType.Male;
        }
    }
}
